pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,434 @@
+                                    );
+                                }
+
+                                show_modal_dialog(props, footerProps);
+                            });
+                            task_proxy.addEventListener("Progress", (event, message) => {
+                                if (/^\.+$/.exec(message) === null) {
+                                    // abrt-retrace-client starts printing dots if the last message it receives is repeated
+                                    this.setState({ message: message, });
+                                }
+                            });
+
+                            this.setState({ task: task_proxy, });
+
+                            task_proxy.Start();
+                        });
+                    });
+        });
+    }
+
+    render() {
+        return <WorkflowRow label={this.state.label}
+                            message={this.state.message}
+                            onCancelButtonClick={this.onCancelButtonClick}
+                            onReportButtonClick={this.onReportButtonClick}
+                            problemState={this.state.problemState}
+                            reportLinks={this.state.reportLinks} />;
+    }
+
+    updateStatusFromBus() {
+        get_problem_property(this.props.problem, "Reports")
+                .done((reports) => {
+                    for (const report of reports[0].v) {
+                        if (!("WORKFLOW" in report[1])) {
+                            continue;
+                        }
+                        if (this.props.workflow[0] !== report[1].WORKFLOW.v.v) {
+                            continue;
+                        }
+                        if (report[0] === "ABRT Server" || report[0] === "uReport") {
+                            continue;
+                        }
+                        const reportLinks = [];
+                        if ("URL" in report[1]) {
+                            reportLinks.push(report[1].URL.v.v);
+                        }
+                        this.setState({
+                            problemState: ProblemState.REPORTED,
+                            reportLinks: reportLinks,
+                        });
+                        break;
+                    }
+                });
+        get_problem_property(this.props.problem, "CanBeReported")
+                .done((reportable) => {
+                    if (!reportable) {
+                        this.setState({ problemState: ProblemState.UNREPORTABLE });
+                    }
+                });
+    }
+}
+
+function WorkflowRow(props) {
+    let status = props.message;
+
+    if (props.problemState === ProblemState.REPORTED) {
+        if (props.reportLinks.length === 1) {
+            status = (
+                <a href={props.reportLinks[0]} rel="noopener noreferrer" target="_blank">
+                    View report <i className="fa fa-external-link" aria-hidden="true" />
```

Comment:
This also needs translation (as well as all other statuses and all action button texts)